src/types/unix_time.ts

```diff
@@ -0,0 +1,27 @@
+export class UnixTime {
+	constructor(private readonly unixTime: number) {
+		if (this.unixTime < 0 || !Number.isInteger(this.unixTime) || !Number.isFinite(this.unixTime)) {
+			throw new Error('Unix time must be a positive integer!');
+		}
```

Comment:
From `src/arfs_entities.ts`:
```ts
export class ArFSEntity {
	...
	unixTime: UnixTime; // seconds since unix epoch, taken at the time of upload, 10 numbers eg. 1620068042
	...
}
```

```suggestion
		if (this.unixTime < 0 || !Number.isInteger(this.unixTime) || !Number.isFinite(this.unixTime)) {
			throw new Error('Unix time must be a positive integer!');
		}
		const length = this.toString().length;
		if (length !== 10) {
				throw new Error(`Unix time must be 10 digits long, got ${length}`);
		}
```

Reply:
So if we do this, we'd need separate types for the two number domains for lastModifiedDate and unixTime...

Reply:
I see... Yeah, we are fine as it is :+1: